app/src/main/java/com/example/rickandmorty/viewModel/MapViewModel.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.rickandmorty.viewModel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.rickandmorty.domain.model.Country
+import com.example.rickandmorty.domain.usecase.GetCountriesUseCase
+import kotlinx.coroutines.flow.*
+
+class MapViewModel(
+    private val getCountriesUseCase: GetCountriesUseCase
+) : ViewModel() {
+
+    val countryFlow = flow<List<Country>> {
+        getCountriesUseCase().fold(
+            onSuccess = {
+                emit(it)
+            },
+            onFailure = {
+                emit(emptyList())
+            }
+        )
```

Comment:
здесь опять же fold это как маппинг, в таком случае лучше тогда делать так
```suggestion
        val state = getCountriesUseCase().fold(
            onSuccess = {
                it
            },
            onFailure = {
                emptyList()
            }
        )
        emit(state)
```
либо как смотрели через onSuccess, onFailure, но есть еще другие функции, можно посмотреть